### PR TITLE
PluginManagerTest: Ensure config keyNames are not duplicated

### PR DESCRIPTION
**Pull request by Nightfirecat**
This prevents runtime errors if an erroneously duplicated keyName is
used in a plugin's configuration.

### CODE REVIEW

File: runelite-client/src/test/java/net/runelite/client/plugins/PluginManagerTest.java [hunk 1]
```diff
@@ -95,6 +98,37 @@ public void before() throws IOException
 			if (pluginDescriptor != null)
 			{
 				pluginClasses.add(clazz);
+				continue;
+			}
+
+			// Ensure plugin configurations do not repeat keyNames
+			if (Config.class.isAssignableFrom(clazz))
```
Comment (deathbeam) [inline runelite-client/src/test/java/net/runelite/client/plugins/PluginManagerTest.java]: You should just build list of config classes here and then make separate test, `@Before` is for preparing environment for tests, not for testing stuff

Comment (Nightfirecat) [reply runelite-client/src/test/java/net/runelite/client/plugins/PluginManagerTest.java]: I've done just that. :+1:

**Comment by ThatGamerBlue** [comment]
what about setting config values from inside the plugin, currently this requires a duplicate key name with a different method signature, from what i can see this test will throw an exception if that case exists

**Comment by deathbeam** [comment]
@ThatGamerBlue nope, because this test has check for method being default

**Pull request merged**